test/e2e/quick_start_test.go

```diff
@@ -0,0 +1,204 @@
+		Expect(err).ToNot(HaveOccurred(), "Invalid configuration. The WORKER_MACHINE_COUNT variable should be a valid integer number")
+
+		By("Running clusterctl config cluster to get a cluster template")
+		workloadClusterTemplate := clusterctl.ConfigCluster(context.TODO(), clusterctl.ConfigClusterInput{
+			// pass reference to the management cluster hosting this test
+			KubeconfigPath: managementClusterKubeConfigPath,
+			// pass the clusterctl config file that points to the local provider repository created for this test,
+			ClusterctlConfigPath: clusterctlConfigPath,
+			// define the cluster template source
+			InfrastructureProvider: e2eConfig.InfraProvider(),
+			// define template variables
+			Namespace:                namespaceName,
+			ClusterName:              clusterName,
+			KubernetesVersion:        e2eConfig.Variables["KUBERNETES_VERSION"],
+			ControlPlaneMachineCount: &controlPlaneMachineCount,
+			WorkerMachineCount:       &workerMachineCount,
+			// setup output path for clusterctl logs
+			LogPath: logPath,
+		})
+		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+
+		By("Applying the cluster template to the cluster")
+		managementCluster.Apply(context.TODO(), workloadClusterTemplate)
+
+		By("Waiting for the cluster infrastructure to be provisioned")
+		cluster = discovery.GetClusterByName(context.TODO(), discovery.GetClusterByNameInput{
+			Getter:    client,
+			Name:      clusterName,
+			Namespace: namespaceName,
+		})
+		Expect(cluster).ToNot(BeNil(), "Failed to get the Cluster object")
+		framework.WaitForClusterToProvision(context.TODO(), framework.WaitForClusterToProvisionInput{
+			Getter:  client,
+			Cluster: cluster,
+		}, e2eConfig.IntervalsOrDefault("quick-start/wait-provision-cluster", "3m", "10s")...)
+
+		By("Waiting for the first control plane machine to be provisioned")
+		controlPlane := discovery.GetKubeadmControlPlaneByCluster(context.TODO(), discovery.GetKubeadmControlPlaneByClusterInput{
+			Lister:      client,
+			ClusterName: clusterName,
+			Namespace:   namespaceName,
+		})
+		Expect(controlPlane).ToNot(BeNil())
+		framework.WaitForOneKubeadmControlPlaneMachineToExist(context.TODO(), framework.WaitForOneKubeadmControlPlaneMachineToExistInput{
+			Lister:       client,
+			Cluster:      cluster,
+			ControlPlane: controlPlane,
+		}, e2eConfig.IntervalsOrDefault("quick-start/wait-provision-first-control-plane-node", "3m", "10s")...)
+
+		By("Installing Calico on the workload cluster")
+		workloadClient, err := managementCluster.GetWorkloadClient(context.TODO(), namespaceName, clusterName)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get the client for the workload cluster with name %s", cluster.Name)
+
+		//TODO: CNI plugin configurable/pluggable (see CAPA)?
+		applyYAMLInput := framework.ApplyYAMLInput{
+			Client:     workloadClient,
+			Scheme:     scheme,
+			Filename:   cniPath,
```

Comment:
different providers could have some minor differences in their Calico file, so maybe make that based on provider

Reply:
Noted. Working ok on this

Reply:
solved